src/View/Widget/DateTimeWidget.php

```diff
@@ -110,25 +109,15 @@ public function render(array $data, ContextInterface $context): string
         }
 
         if (!isset($data['step'])) {
-            $step = $this->defaultStep[$data['type']];
+            $data['step'] = $this->defaultStep[$data['type']];
```

Comment:
Shouldn't this default step go in the magic setStep() function too?

Reply:
The `setStep()` has `$fieldName` as an argument so I called it only if `$data['fieldName']` is set.

Also I got rid of the trait since put it methods directly in `BasicWidget` since all other widgets extend it anyway. Also `setStep()` is now overridden in `DateTimeWidget()` so introspection related to datetime fields is contained within same class.

Reply:
It looks good the way it is now.  I approved.